tests/checks/main-branch.nf/.checks

```diff
@@ -0,0 +1,6 @@
+#
+# run remote pipeline, which has default branch 'main'
+#
+$NXF_CMD -q run nextflow-io/socks $WITH_DOCKER --out result.txt | tee stdout
+[[ `grep 'Launching \`https://github.com/nextflow-io/socks\`' .nextflow.log | grep -c '[main]'` == 1 ]] || false
+[[ $? == 0 ]] || exit 1
```

Comment:
Let's keep this as a unit test. Something like this 

https://github.com/nextflow-io/nextflow/blob/e794e868fd3d62346e09e5deeba612976dd5143e/modules/nextflow/src/test/groovy/nextflow/scm/AssetManagerTest.groovy#L160-L178

Reply:
My initial attempt was along these lines, but downloading isn't enough to trigger the "stuck on revision" error - it only happens when trying to run the pipeline.

Reply:
It can be triggered by calling two methods:
```
manager.download()
manager.checkout(null)
```

but is that perhaps making too many assumptions about how the code will executed when actually running?

Reply:
> but downloading isn't enough to trigger the "stuck on revision" error

is that really needed? at the end the test should verify that the repo is cloned with the expected  (non-default) branch (i guess)

Reply:
The repo is successfully cloned - even before these changes - and jgit already automatically checks out the remote's default branch (in the socks case, `main`).

The issue happens later: when running the pipeline without specifying a revision, nextflow checks that the current branch matches what it expects the default branch to be (ie `master`).

Reply:
I see, then let's go with [this](https://github.com/nextflow-io/nextflow/pull/5375#discussion_r1791872286) approach, no?

Reply:
Updated. 

It's a simpler test this way, but it does carry a small risk of failing to catch a regression if the real codepath is modified to work differently.

Reply:
Excellent